deploy/src/utils.rs

```diff
@@ -0,0 +1,28 @@
+use anyhow::Result;
+use std::cmp::Ordering;
+
+/// Compare two CRD version
+pub(crate) fn compare_versions(version1: &str, version2: &str) -> Result<Ordering> {
+    let v1_parts: u32 = version1.trim_start_matches('v').parse()?;
+    let v2_parts: u32 = version2.trim_start_matches('v').parse()?;
```

Comment:
I think the str could be directly compared without converting to number types, as they all start with 'v'?

Reply:
Could this result in v2 being greater than v10? By the way, since the API version can be written as 'v1beta' or similar, this function will be updated as more API versions are added.